mllib/src/main/scala/org/apache/spark/ml/classification/ProbabilisticClassifier.scala

```diff
@@ -201,11 +201,24 @@ abstract class ProbabilisticClassificationModel[
       probability.argmax
     } else {
       val thresholds: Array[Double] = getThresholds
-      val scaledProbability: Array[Double] =
-        probability.toArray.zip(thresholds).map { case (p, t) =>
-          if (t == 0.0) Double.PositiveInfinity else p / t
+      val probabilities = probability.toArray
```

Comment:
This code was implemented in the PR for MLOR. I included it here since it is more generic. It requires only a single pass through the probabilities array using a while loop, so it should be slightly faster.